### ISSUE TITLE
Week 3 Discussion: Market Problem Statement

### ISSUE THREAD

**Issue by kayvee** [stefanbund/cis155#9]
Hello everyone. I'm starting off the discussion for this week again.

**What things must you do next, to set yourself up for success?**

I think the first thing I need to accomplish is finishing Phase 1! For me, at the moment, this includes:
- creating a hash (or dictionary, or ArrayList, etc, any data structure essentially, depending on how I decide to do it) and using random to select industries and symbols for the stocks
- complete the buildRandomPortfolio function
- comment the code more thoroughly to make sure no 'parts' are missing and to be positive I understand the mechanics at work in the code

**What things in your skillset are missing, or need to be present to help contribute to the end solution?**
- I need to review how to best utilize random with a specific data structure for selecting industries and creating symbols. I would rather do this as a more simple and cleaner approach than 20+ if/else if statements.

**What strategies are you envisioning, as necessary to complete the project at the end of the month?**
- At this point, I don't think I have specific strategies other than breaking through the code with more comments to explain the logic to myself. I think I understand what's been done so far, and at the moment, I'm already done with all of the posted videos, so I can spend these next 2-3 weeks working on the project itself.
## 

Side note: Phase 1 is due at the end of the month, correct, with Milestone 2 due at the end of next week? The Market is listed as Phase 2, so it will be due 5 weeks after Phase 1, if I understood the course materials correctly.

**Comment by forrestbennett** [comment]
**What things must you do next, to set yourself up for success?**
I need to go back and watch the videos again. I have the correct outputs and my code is without errors but how I got there seems like a distant memory. Now that I have the first milestone complete it will help to see it all again and connect the dots.
**What things in your skillset are missing, or need to be present to help contribute to the end solution?**
Again, watching the videos after completing milestones to fully comprehend the process.
**What strategies are you envisioning, as necessary to complete the project at the end of the month?**
I need to leave more comments for reference for myself.

**Comment by kayvee** [comment]
@stefanbund 

Re: Milestone 2 and Phase 1

Will specifics be posted again as they were for Milestone 1 (what videos we should watch and what we should submit)?

**Comment by stefanbund** [comment]
Yes, coming up ASAP

From Stefan

**Comment by stefanbund** [comment]
Just replying to your question on the milestones. Specifics will be posted, to help you along. You are correct in your comment, however. I tried to consolidate instructions for the assignment within that master course document. Is that easy enough to understand?

From Stefan

**Comment by kayvee** [comment]
Absolutely!

I just wanted to make sure that the milestone would at have something
posted since it is a little bit more based on the videos.

The project document + the videos make the phase 1 very clear.

Thank you!

On Fri, Feb 19, 2016 at 10:59 AM, stefan bund <notifications@github.com
<javascript:_e(%7B%7D,'cvml','notifications@github.com');>> wrote:

> Just replying to your question on the milestones. Specifics will be
> posted, to help you along. You are correct in your comment, however. I
> tried to consolidate instructions for the assignment within that master
> course document. Is that easy enough to understand?
> 
> From Stefan
> 
> > On Feb 17, 2016, at 12:27 AM, Kaveh Varjoy <notifications@github.com
> > <javascript:_e(%7B%7D,'cvml','notifications@github.com');>> wrote:
> > 
> > Hello everyone. I'm starting off the discussion for this week again.
> > 
> > What things must you do next, to set yourself up for success?
> > I think the first thing I need to accomplish is finishing Phase 1! For
> > me, at the moment, this includes:
> > 
> > creating a hash (or dictionary, or ArrayList, etc, any data structure
> > essentially, depending on how I decide to do it) and using random to select
> > industries and symbols for the stocks
> > complete the buildRandomPortfolio function
> > comment the code more thoroughly to make sure no 'parts' are missing and
> > to be positive I understand the mechanics at work in the code
> > 
> > What things in your skillset are missing, or need to be present to help
> > contribute to the end solution?
> > 
> > I need to review how to best utilize random with a specific data
> > structure for selecting industries and creating symbols. I would rather do
> > this as a more simple and cleaner approach than 20+ if/else if statements.
> > 
> > What strategies are you envisioning, as necessary to complete the
> > project at the end of the month?
> > 
> > At this point, I don't think I have specific strategies other than
> > breaking through the code with more comments to explain the logic to
> > myself. I think I understand what's been done so far, and at the moment,
> > I'm already done with all of the posted videos, so I can spend these next
> > 2-3 weeks working on the project itself.
> > Side note: Phase 1 is due at the end of the month, correct, with
> > Milestone 2 due at the end of next week? The Market is listed as Phase 2,
> > so it will be due 5 weeks after Phase 1, if I understood the course
> > materials correctly.
> > 
> > —
> > Reply to this email directly or view it on GitHub.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/stefanbund/cis155/issues/9#issuecomment-186361814.

## 

Sent from my iPhone

**Comment by stefanbund** [comment]
Thanks for your encouragement about the videos, there will be a lot more for the next project.

From Stefan

**Comment by kayvee** [comment]
Sorry for more logistical posts, but will the Dropbox for Milestone 2, Post 3, and Post 4 be added to Seaport?

**Comment by Irfan009** [comment]
Q: What things must you do next, to set yourself up for success?
A: Work more on Methods, Arrays, and Logical Statements (if, else)

Q: What things in your skillset are missing, or need to be present to help contribute to the end solution?
A: I feel my biggest area of improvement, while working on a problem is choosing best option available out of different possible available choices. I am struggling with grasping the concepts of method. 

Q: What strategies are you envisioning, as necessary to complete the project at the end of the month?
A: To be successful in learning a programming language I believe the key lies in practice, repetition and experimentation and that’s what I am planning on doing.

~Irfan Awan

**Comment by stefanbund** [comment]
Yes

From Stefan

**Comment by stefanbund** [comment]
You are right that managing collections and other ways to capture many items in one set will be increasingly valuable.

From Stefan

**Comment by WesleyTo** [comment]
**_What things must you do next, to set yourself up for success?_**
Keep watching the videos and reading the books. Also, looking up things in the Java documentation really helps as far as getting low-level understanding of the language.
[https://docs.oracle.com/javase/8/docs/api/](url)

**_What things in your skillset are missing, or need to be present to help contribute to the end solution?_**
I'm still working out some haziness on the Random class. I also want to get a grasp on OOP, maybe design patterns, such that I can more easily think about the design from a high level.

**_What strategies are you envisioning, as necessary to complete the project at the end of the month?_**
In my experience so far, thinking about the solution is 50% of the work. 10% is the actual coding, and the other 40% is rethinking and debugging. Allocating the required time for each part (general time management) will be vital to project completion.

**Comment by WesleyTo** [comment]
@Irfan009 
I'm wondering what you mean by " I am struggling with grasping the concepts of method."

Do you mean OOP in general? Or methods/functions in and of themselves?

**Comment by forrestbennett** [comment]
@WesleyTo along the same lines I like to lookup "Java programming" on youtube. Some are super advanced but others are basic and help a lot. Not that Stefen's vids are not enough, its sometimes helpful to see it done by others as well.

**Comment by pricedjp** [comment]
Q: What things must you do next, to set yourself up for success?
A: Spend more time focusing on writing clean code and working on the structure of logical statements

Q: What things in your skillset are missing, or need to be present to help contribute to the end solution?
A: Getting to a point where my understanding of structure of variables, arrays, and on a more micro level – proper use of bracketing and spacing. 

Q: What strategies are you envisioning, as necessary to complete the project at the end of the month?
A: Time, re-watching videos, utilizing the weekly discussion forums to post errors in my code and get assistance when I can’t find a solution myself. 
Dustin Price

**Comment by stefanbund** [comment]
This is the first time I've done stuff with YouTube. I find my students are already looking things up on that site. Thus I hope to participate in that culture.

From Stefan

**Comment by stefanbund** [comment]
I am agreeing with folks who see programming as a very mental game. You guys will have a pretty solid grasp of the language in a few weeks, so it will come down to a sense of strategy and architecture.

From Stefan

**Comment by forrestbennett** [comment]
@stefanbund Life existed and everyone managed to get by before youtube was a thing, but its hard to think of life without it! I use youtube almost daily, it is a powerful tool. Its awesome you are able to tap in and utilize it within your career.

**Comment by calvinln** [comment]
Q: What things must you do next, to set yourself up for success?
I think I need to be more comfortable with eclipse, java, and writing code in general.

Q: What things in your skillset are missing, or need to be present to help contribute to the end solution?
I need to learn how to make collections of objects and manipulate them.

Q: What strategies are you envisioning, as necessary to complete the project at the end of the month?
Watching videos on youtube, reading the book, getting more familiar with eclipse, and keep coding.

Calvin

**Comment by trungphung** [comment]
Hi everyone,
1) What things must you do next, to set yourself up for success?
- I am still very confuse about the language, so like many others, have to re watch the videos.  I think I am gonna draw my visualization of the structure on a piece of paper 

2) What things in your skill set are missing, or need to be present to help contribute to the end solution?
- I still haven't been able to create a visualization in my mind 

3)What strategies are you envisioning, as necessary to complete the project at the end of the month?
- I must start one line at a time

Trung Phung

**Comment by khaleb07** [comment]
What things must you do next, to set yourself up for success?
View the videos and read the text more frequently to acquire greater knowledge of the subjects at hand.

What things in your skillset are missing, or need to be present to help contribute to the end solution?
Better knowledge of arrays and developing the structure of the program.

What strategies are you envisioning, as necessary to complete the project at the end of the month?
I definitely need to input more hours into this class, I have been slacking off majorly.

**Comment by kayvee** [comment]
@trungphung : you mention visualizations and the language. Which part of this is giving you problem?

Is it the actual assignment of classes? Or is it the flow of the language? I think that they have very similar mechanisms behind them but they also have their own quirks. The best way, I think, to think about it is the following:

Imagine you want to bake a cake. You know that there are multiple parts. Nothing is related to each other necessarily. The mixing bowl doesn't need the oven. The oven doesn't need the washing machine. But wait, the washing machine does need the mixing bowl for afterwards! However, the mixing bowl will never need the washing machine itself. So you draw arrows 'from object to object.'

That sets up our classes. We know our parts and how they relate to each other. Now we have to ask what each object _does_ by (and also "knows about")  itself, and what it does with (and "knows of") other objects.

The former are the variables and functions defined in the class. The latter are the functions provided by other classes, as well as making sure to make available information from that object when necessary.

**Comment by stefanbund** [comment]
Classes make for some challenges when learning the language. 

I like to think of classes as storage locations for certain pieces of data. Since classes can only contain a list of variables and a list of methods, it helps explain classes as storage places for data,  which also have the ability to do certain things.

Classes also help the programmer organize his or her thought process. They also help the programmer model the program after things in the real world. This allows for more flow inside the main and other classes which may draw upon classes.

Overall, classes help the programmer as much as they enable efficiency within the application.

From Stefan

**Comment by pricedjp** [comment]
What things must you do next, to set yourself up for success?
The keys to success are patiently watching the videos. Pausing and rewinding the videos to make sure I am grasping the concepts and allowing myself to play with whats being taught. Consistently making errors, understanding why I made the error,and fixing the code. 

What things in your skill-set are missing, or need to be present to help contribute to the end solution?
Coding is foreign to me. Understanding how code in the main is interacting with the code I'm writing in my brain center and stock interact is new to me. 

What strategies are you envisioning, as necessary to complete the project at the end of the month?
Continuing to persevere through the videos.

**Comment by stefanbund** [comment]
Understood. I have found teaching at the Cal State University level, students feel like they are getting a lot of information dumped on them, at this stage. Once you are finished with this phase, however, you will see the learning curve flatten, and your ability to solve many many problems using the same toolset. I understand at this moment what you are experiencing, but the good news is it gets better.

From Stefan

**Comment by Sweeter-man** [comment]
What things must you do next, to set yourself up for success?
become more familiarized with the syntax and architecture,  repetition. 

What things in your skill-set are missing, or need to be present to help contribute to the end solution?
understanding everything is grouped or has a connection somewhere else is the code,
and being able to understand all the connections. 

What strategies are you envisioning, as necessary to complete the project at the end of the month?
Repetition in order to understand the syntax so that way I can build the code better.

**Comment by alsinha** [comment]
What is video player that plays well, I am having issues sometimes playing the video

**Comment by forrestbennett** [comment]
@alsinha The videos play on youtube. If it is not playing you might need an update.

**Comment by stefanbund** [comment]
Also, if you are using an Apple, flash may be disabled if you're using a power saving techniques. You might check your Apple preferences to get around this

From Stefan
